code/modules/tgui/tutorial/tgui_template_machine.dm

```diff
@@ -0,0 +1,62 @@
+/obj/machinery/tgui_template
+	name = "tgui template machine"
+	desc = "Coders' helper."
+	icon = 'icons/obj/hydroponics/equipment.dmi'
+	icon_state = "sextractor_open"
+
+	var/list/options
+	var/ui_message
+	var/current_option
+
+/obj/machinery/tgui_template/Initialize(mapload)
+	. = ..()
+
+	options = list("Love", "Bravery", "Honor")
+	ui_message = "Hello, world!"
+	current_option = options[1]
+
+/obj/machinery/tgui_template/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "TGUITemplate")
+		ui.open()
+
+
+/obj/machinery/tgui_template/ui_data(mob/user)
```

Comment:
This needs explanation about how without auto updating, this won't react to external changes. If this is to react to external changes, for example from a signal firing, then that signal needs to call ui_update in order to mark all of the open UIs for this instance as dirty.